packages/astro/src/vite-plugin-astro-server/plugin.ts

```diff
@@ -43,6 +44,16 @@ export default function createVitePluginAstroServer({
 			viteServer.watcher.on('unlink', rebuildManifest.bind(null, true));
 			viteServer.watcher.on('change', rebuildManifest.bind(null, false));
 
+			function handleUnhandledRejection(rejection: any) {
+				const error = new AstroError({
+					...AstroErrorData.UnhandledRejection,
+					message: AstroErrorData.UnhandledRejection.message(rejection?.stack || rejection)
+				})
+				logger.error(null, error.message);
+			}
```

Comment:
It's a bit of a edge case, but presumably this should pass through our normal-ish error pipeline so we get a nicely formatted error. How does this look in the browser too?

Reply:
It can't pass through the normal error pipeline, it happens outside of the context of a request. You could do: `setTimeout(() => Promise.reject('foo'), 5000)` and there is no ongoing request, the rejection still happens.

It doesn't currently show up in the browser, do we have a way to send an error from the server to the client outside of a request?

Reply:
I would think that from where you are in the code, you'd be able to send to the websocket since you're so close to the server creation, here's how it works in the rendering: https://github.com/withastro/astro/blob/f8b27b07c064e84a9b0c51c16fe7fdb72b2ef5f6/packages/astro/src/vite-plugin-astro-server/response.ts#L32

You should just need to send the error payload through the websocket and it should work!»

> It can't pass through the normal error pipeline, it happens outside of the context of a request. 

If you can't make it go through the entire pipeline, at least formatting the error message would be great: https://github.com/withastro/astro/blob/f8b27b07c064e84a9b0c51c16fe7fdb72b2ef5f6/packages/astro/src/core/messages.ts#L257

Reply:
I think AsyncLocalStorage can maybe allow this, going to give that try.

Reply:
And so it begins... I have _many_ ideas for AsyncLocalStorage!

Reply:
Updated the PR and it should work now. Updated with a screenshot of what the browser receives. Would appreciate another review @Princesseuh